src/lexer/regexp.ts

```diff
@@ -142,10 +142,15 @@ export function scanRegularExpression(parser: ParserState, context: Context): To
  */
 function validate(parser: ParserState, pattern: string, flags: string): RegExp | null | Token {
   try {
-    // Temporarily allows older version of Nodejs (or browser) to accept the new Indices flag "d".
-    flags = flags.replace('d', '');
     return new RegExp(pattern, flags);
   } catch (e) {
-    report(parser, Errors.UnterminatedRegExp);
+    try {
+      // Some JavaScript engine has not supported flag "d".
+      new RegExp(pattern, flags.replace('d', ''));
```

Comment:
Could not just fallback to `new RegExp(pattern)`. Some flag affects the correctness of the pattern like `/\1/u` is incorrect, but `/\1/` is correct.

So we only anticipate missing "d" support here.